.github/workflows/flake8.yml

```
@@ -0,0 +1,27 @@
+name: Python Flake8 Linter
+on:
+  pull_request:
+jobs:
+  flake8-linter:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout source
+        uses: actions/checkout@v3
+      - name: Install Python 3.11
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.11'
+      - name: Install Flake8 5.0.4 linter
+        run: pip install flake8==5.0.4  # use this version for --diff option
+      - name: Setup Flake8 output matcher for PR annotations
+        run: echo '::add-matcher::.github/workflows/flake8-linter-matcher.json'
+      - name: Fetch pvlib main branch
+        run: |
+          git remote add upstream https://github.com/pvlib/pvlib-python.git
+          git fetch upstream main
+      - name: Run Flake8 linter
+        run: git diff upstream/main HEAD | flake8
```

```suggestion
        run: git diff upstream/$GITHUB_BASE_REF HEAD | flake8
```

Should we do this so that the diff is always taken with respect to the branch the PR is opened against?  We rarely (never?) open PRs to any branch except `main`, so maybe doesn't matter, but I could see us using release branches and such at some point in the future.

Docs: https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables